truffle/com.oracle.truffle.api.debug/src/com/oracle/truffle/api/debug/Debugger.java

```diff
@@ -957,15 +957,9 @@ public FrameInstance visitFrame(FrameInstance frameInstance) {
                     if (stackIndex < contextStackDepth) {
                         final Node callNode = frameInstance.getCallNode();
                         if (callNode != null) {
```

Comment:
I'd say this check is now useless. 
You can push frames.add(frameInstance) just before stackIndex++.